website/source/docs/platform/k8s/connect.html.md

```diff
@@ -0,0 +1,259 @@
+
+### Accepting Inbound Connections
+
+An example pod is shown below with Connect enabled to accept inbound
+connections. Notice that the pod would still be fully functional without
+Connect. Minimal to zero modifications are required to pod specifications to
+enable Connect in Kubernetes.
+
+This pod specification starts an server that responds to any
+HTTP request with the static text "hello world".
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: static-server
+  annotations:
+    "consul.hashicorp.com/connect-inject": "true"
+spec:
+  containers:
+    - name: static-server
+      image: hashicorp/http-echo:latest
+      args:
+        - -text="hello world"
+        - -listen=:8080
+      ports:
+        - containerPort: 8080
+          name: http
+```
+
+The only change for Connect is the addition of the
+`consul.hashicorp.com/connect-inject` annotation. This enables injection
+for this pod. The injector can also be
+[configured](/docs/platform/k8s/connect.html#installation-and-configuration)
+to automatically inject unless explicitly disabled, but the default
+installation requires opt-in using the annotation shown above.
+
+This will start a Connect sidecar that listens on a random port registered
+with Consul and proxies valid inbound connections to port 8080 in the pod.
+To establish a connection to the pod, a client must use another Connect
+proxy. The client Connect proxy will use Consul service discovery to find
+all available upstream proxies and their public ports.
+
+In the example above, the server is listening on `:8080`. This means
+the server will still bind to the pod IP and allow external connections.
+This is useful to transition to Connect by allowing both Connect and
+non-Connect connections. To restrict access to only Connect-authorized clients,
+any listeners should bind to localhost only (such as `127.0.0.1`).
+
+### Connecting to Connect-Enabled Services
+
+The example pod specification below configures a pod that is capable
+of establishing connections to our previous example "static-server" service. The
+connection to this static text service happens over an authorized and encrypted
+connection via Connect.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: static-client
+  annotations:
+    "consul.hashicorp.com/connect-inject": "true"
+    "consul.hashicorp.com/connect-service-upstreams": "static-server:1234"
+spec:
+  containers:
+    - name: static-client
+      image: tutum/curl:latest
+      # Just spin & wait forever, we'll use `kubectl exec` to demo
+      command: [ "/bin/sh", "-c", "--" ]
+      args: [ "while true; do sleep 30; done;" ]
+```
+
+Pods must specify upstream dependencies with the
+[`consul.hashicorp.com/connect-service-upstreams` annotation](/docs/platform/k8s/connect.html#consul-hashicorp-com-connect-service-upstreams).
+This annotation declares the names of any upstream dependencies and a
+local port to listen on. When a connection is established to that local
```

Comment:
It'll just be a race like any other port conflict, nothing unusual. One will bind and the other will fail. Not sure if we need to go into that detail since its just standard unix...